### ISSUE TITLE
Add support to get only metadata from the triggers

### ISSUE THREAD

**Issue by paulbatum** [Azure/azure-functions-host#2698]
_From @pragnagopa on April 17, 2018 17:49_

Please provide a succinct description of the issue.

Scenario requested in [Azure/azure-functions-host/issues/2301](https://github.com/Azure/azure-functions-host/issues/2301)

When binding to a CloudBlockBlob in a C# function,  user can control when to read stream attached to the binding and if needed can be skip it completely.

It is worth adding this support for other languages in V2 as  language workers as of now only have support to bind to basic data types: bytes, string and json

_Copied from original issue: Azure/azure-webjobs-sdk#1667_

**Comment by JonathanGiles** [comment]
I wholeheartedly endorse this issue! It leads to issues as discussed in https://github.com/Azure/azure-functions-java-library/issues/42 where I don't actually even want the data - I just want to know that a new file has landed.

**Comment by brettsam** [comment]
This is marked with v3.x but I don't believe it's on the table and likely needs to be punted. @fabiocav / @paulbatum thoughts?

**Comment by paulbatum** [comment]
I agree. Also if we already have issues for the rich types featureset, then this one might be a dupe.

**Comment by Josh-a-e** [comment]
Any progress on this?

**Comment by aangelisc** [comment]
I'd also like to know if there is any progress on this. The Event Grid workaround doesn't fit my solution as we're operating function apps within a VNET. It'd be much simpler being able to retrieve the blob metadata and then handle blob content without the gRPC restrictions.

**Comment by Josh-a-e** [comment]
My solution here is just to publish my own event when I upload the files, but it's a shame to not have this handled natively.
